### DISCUSSION TITLE
Changing font size depending on external monitor (work vs home) or built in display

### DISCUSSION THREAD

**Discussion by rameshms-work** [kovidgoyal/kitty#6864] [Q&A]
At work i have a bigger, high res monitor and therefore I prefer to use a bigger font.
While at home, i have a lower res monitor and therefore I prefer to use a smaller font.
Currently, I have to edit the font size in config file manually every time I switch kitty to different displays and applying the config.

Is there a way to automate this ?

OS: OSX 14.1.1
Mac book pro 16

**Comment by kovidgoyal** [comment]
Font size is specified in points which is a physical size, so its
independent of monitor resolutions. As such there is no way to specify
different per monitor sizes. You can use the include mechanism in
kitty.conf to have per computer config, based on environmnet variables.

**Comment by benfrain** [comment]
I have this requirement too, same laptop but gets used with a different Monitor depending on whether working from office or home. 

I tried to do this in macOS but not working. I believe it is due to the var not being available in Kitty?

I set this in my .zshrc file (first find out whether a studio display is attached, and if so set the var accordingly):

```
if system_profiler SPDisplaysDataType | grep -q "Studio Display"; then export DISPLAYNAME="studiodisplay"; fi
```

Which sets that variable to "studiodisplay", provable with `printenv`. 

I then have a suitable `studiodisplay.conf` file containing `font_size 22`.

In my `kitty.conf` I have this:

```
include ${DISPLAYNAME}.conf
```

However, Kitty is not seeing the variable, provable by running `show_kitty_env_vars` from Kitty.

Presume this is because Kitty already runs before is 'reads' that value? If so, is it solvable by setting it earlier somehow?

**Comment by kovidgoyal** [comment]
env vars for kitty cannot be set in shell scripts shell scripts are not
read till after kitty starts. You need to define it system wide, see
https://sw.kovidgoyal.net/kitty/faq/#things-behave-differently-when-running-kitty-from-system-launcher-vs-from-another-terminal

**Reply by benfrain** [reply]
OK, thought for a moment that it might be possible to do this in the kitty conf file but expect if/else in config would slow things considerably.

```
if system_profiler SPDisplaysDataType | grep -q "Studio Display"; then 
font_size 16
else
font_size 19
fi
```

**Reply by kovidgoyal** [reply]
yes, kitty's config language is deliberately kept simple and static. It would be easy to allow dynamic configs via say a python script, but that has other tradeoffs, like making configs slower/harder to debug/validate. I dont use macOS so I havent kept up with the latest way to set systemwide env vars in macOS is, Apple keeps changing these every few releases.
